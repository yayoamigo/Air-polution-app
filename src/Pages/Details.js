import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';

const Details = () => {
  const { name } = useParams();
  const countriesArr = useSelector((state) => state.countries.countries);
  const isLoading = useSelector((state) => state.countries.isLoading);
  const filterCountry = countriesArr.filter((country) => country.Name === name);
  const { components } = filterCountry[0].list[0];

  return (
    <div>
      {isLoading ? (
        <div className="loading-container">
          <div className="loading-wheel" />
          <p className="loading-text">Please wait</p>
        </div>
      ) : (
        <div className="details">
          <h2>{filterCountry[0].Name}</h2>
          {Object.entries(components).map(([key, value]) => (
            <div key={key} className="components">
              <h3>{key}</h3>
              <p>{value}</p>
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

export default Details;
