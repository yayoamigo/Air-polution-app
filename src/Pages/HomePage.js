import React, {useEffect} from 'react'
import { Nav } from '../Components/Nav';
import { FilterBy } from '../Components/FilterBy';
import { useSelector, useDispatch } from 'react-redux';
import { getValues } from '../Redux/ducks/slices';
import Countries from '../Components/Countries';
import Search from '../Components/Search';


export const HomePage = () => {
 const countriesArr = useSelector((state) => state.countries.countries);
 const isLoading = useSelector((state) => state.countries.isLoading);
 const selectedContinent = useSelector((state) => state.countries.selectedContinent)
 const { length} = countriesArr;
 const dispatch = useDispatch();
 useEffect(() => {
  if (length === 0) {
   dispatch(getValues())
  }
 }, [dispatch]);

 const filteredCountries = countriesArr.filter((country) => {
  return country.Continent === selectedContinent
})

  return (
   <div className='App'>
   <Nav/>
   <div className='filters'>
    <Search/>
    <FilterBy />
   </div>
   <div className='Box'>
   {isLoading ? (
  <p>Is loading</p>
) : (
  filteredCountries.map((country)=>(
    <Countries
    key={country.Name}
    name = {country.Name}
    aqi = {country.list[0].main.aqi}
    continent = {country.Continent}
    />
  ))
)}
   </div>
  </div>
  )
}
