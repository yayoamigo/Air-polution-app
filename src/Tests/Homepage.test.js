import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HomePage from '../Pages/HomePage';
import Details from '../Pages/Details';
import store from '../Redux/configureStore';

it('render', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});


