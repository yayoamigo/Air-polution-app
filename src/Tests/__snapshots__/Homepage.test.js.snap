// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="App"
>
  <div
    className="filters"
  >
    <div
      className="searchdiv"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="search..."
        type="text"
        value=""
      />
    </div>
    <div
      className="filterDiv"
    >
      <p
        className="filterTitle"
      >
        Filter by continent
      </p>
      <select
        className="selectdiv"
        onChange={[Function]}
        value="All"
      >
        <option
          className="select"
          value="All"
        >
           All
        </option>
        <option
          className="select"
          value="Africa"
        >
           Africa
        </option>
        <option
          className="select"
          value="Europe"
        >
           Europe
        </option>
        <option
          className="select"
          value="North America"
        >
           North America
        </option>
        <option
          className="select"
          value="Asia"
        >
           Asia
        </option>
        <option
          className="select"
          value="South America"
        >
           South America
        </option>
        <option
          className="select"
          value="Oceania"
        >
           Oceania
        </option>
      </select>
    </div>
  </div>
  <div
    className="Box"
  >
    <div
      className="loading-container"
    >
      <div
        className="loading-wheel"
      />
      <p
        className="loading-text"
      >
        Please wait
      </p>
    </div>
  </div>
</div>
`;
